src/main/java/com/ibm/cloud/sdk/core/service/security/IamTokenManager.java

```diff
@@ -181,14 +183,32 @@ private boolean isRefreshTokenExpired() {
    * @return object containing requested IAM token information
    */
   private IamToken callIamApi(Request request) {
-    Call call = HttpClientSingleton.getInstance().createHttpClient().newCall(request);
-    ResponseConverter<IamToken> converter = ResponseConverterUtils.getObject(IamToken.class);
+    final IamToken[] returnToken = new IamToken[1];
```

Comment:
Does this work in Android? I think their have their own Thread management system

Reply:
Just managed to test it out and it looks like Android accepts the Java threading just fine 👌